common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftTokenErrorResponse.java

```diff
@@ -0,0 +1,54 @@
+package com.microsoft.identity.common.internal.providers.microsoft;
+
+import com.google.gson.annotations.SerializedName;
+import com.microsoft.identity.common.internal.providers.oauth2.TokenErrorResponse;
+
+import java.util.ArrayList;
+
+public class MicrosoftTokenErrorResponse extends TokenErrorResponse {
+
+    @SerializedName("error_codes")
+    protected ArrayList<Long> mErrorCodes;
+
+    @SerializedName("timestamp")
+    protected String mTimeStamp;
+
+    @SerializedName("trace_id")
+    protected String mTraceId;
+
+    @SerializedName("correlation_id")
+    protected String mCorrelationId;
+
+
+    public ArrayList<Long> getErrorCodes() {
```

Comment:
Unless there's a specific reason to return a concrete type here, I'd prefer if this returned `List<Long>` rather than `ArrayList<Long>`